File: coding/java/BinarySearch.java

import java.util.Scanner;

class BinarySearch {
    public static void main(String[] args) {
        int i, mid, first, last, x, n;
        boolean found = false;

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of elements: ");
        n = sc.nextInt();

        int a[] = new int[n];

        System.out.println("Enter elements of array (in sorted order): ");
        for (i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }

        System.out.print("Enter element to search: ");
        x = sc.nextInt();

        first = 0;
        last = n - 1;

        while (first <= last) {
            mid = (first + last) / 2;

            if (a[mid] == x) {
                found = true;
                System.out.println("Element found at position: " + (mid + 1));
                break;
            } 
            else if (a[mid] > x) {
                last = mid - 1;
            } 
            else {
                first = mid + 1;
            }
        }

        if (!found) {
            System.out.println("Element not found");
        }
    }
}

File: coding/PYTHON/DSA-PYTHON/Sky/binary_search.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] < target:
            low = mid + 1
        elif arr[mid] > target:
            high = mid - 1
        else:
            return mid  # Return index of target element

    return -1  # Return -1 if target is not in the list


# Get input from the user and sort the list
arr = list(map(int, input("Enter elements of the sorted list separated by space: ").split()))
target = int(input("Enter the target element to search for: "))
result = binary_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print(f"Element {target} not found in the list")
